instructor/function_calls.py

```diff
@@ -249,6 +247,22 @@ def parse_gemini_json(
             # Pydantic non-strict: https://docs.pydantic.dev/latest/concepts/strict_mode/
             return cls.model_validate(parsed, context=validation_context, strict=False)
 
+    @classmethod
+    def parse_gemini_tools(
+        cls: type[BaseModel],
+        completion: ChatCompletion,
+        validation_context: Optional[dict[str, Any]] = None,
+        strict: Optional[bool] = None,
+    ) -> BaseModel:
+        tool_call = completion.parts[0].function_call
+        assert (
+            tool_call.name == cls.openai_schema["name"]  # type: ignore[index]
+        ), "Tool name does not match"
+        # reference https://ai.google.dev/gemini-api/tutorials/extract_structured_data
+        response = type(tool_call).to_dict(tool_call)["args"]
+
+        return cls.model_validate(response, context=validation_context, strict=strict)
```

Comment:
@jxnl i've been playing around with this and it seems like perhaps we might want to set strict to be False by default for gemini tools?

Let's take the simple example below

```
import instructor
import google.generativeai as genai
from pydantic import BaseModel

client = instructor.from_gemini(
    genai.GenerativeModel("gemini-1.5-pro"),
    mode=instructor.Mode.GEMINI_TOOLS,
)


class Weather(BaseModel):
    location: str
    hour: int
    temperature: float


resp = client.chat.completions.create(
    response_model=Weather,
    messages=[
        {
            "role": "user",
            "content": "The weather in Singapore at 5pm is 34.5 degree celcius",
        },
    ],
    strict=False,
)
```

When we process this in `process_response`, we get an error if strict isn't true. This seems to happen because when we coerce the protobuf to a json dictionary, it coerces all the ints into floats for some reason.

```python
@classmethod
    def parse_gemini_tools(
        cls: type[BaseModel],
        completion: ChatCompletion,
        validation_context: Optional[dict[str, Any]] = None,
        strict: Optional[bool] = None,
    ) -> BaseModel:
        tool_call = completion.parts[0].function_call
	print(tool_call)
        assert (
            tool_call.name == cls.openai_schema["name"]  # type: ignore[index]
        ), "Tool name does not match"
        # reference https://ai.google.dev/gemini-api/tutorials/extract_structured_data
        response = type(tool_call).to_dict(tool_call)["args"]
	print(response)

        return cls.model_validate(response, context=validation_context, strict=strict)
```

This in turn gives u the following output when we print out the value at each stage

```
name: "Weather"
args {
  fields {
    key: "temperature"
    value {
      number_value: 34.5
    }
  }
  fields {
    key: "location"
    value {
      string_value: "Singapore"
    }
  }
  fields {
    key: "hour"
    value {
      number_value: 17
    }
  }
}

{'hour': 17.0, 'location': 'Singapore', 'temperature': 34.5}
location='Singapore' hour=17 temperature=34.5
```

Reply:
I realised that we're doing that already for the vertexai_tools so that's not entirely inconsistent with the existing approach. Then we can throw an error if users set strict=True in the function call

```python
@classmethod
    def parse_vertexai_tools(
        cls: type[BaseModel],
        completion: ChatCompletion,
        validation_context: Optional[dict[str, Any]] = None,
        strict: Optional[bool] = None,
    ) -> BaseModel:
        strict = False
        tool_call = completion.candidates[0].content.parts[0].function_call.args  # type: ignore
        model = {}
        for field in tool_call:  # type: ignore
            model[field] = tool_call[field]
        return cls.model_validate(model, context=validation_context, strict=strict)
```